src/mame/nintendo/nes.cpp

```diff
@@ -121,6 +121,20 @@ void nes_state::famicom(machine_config &config)
 	SOFTWARE_LIST(config, "cass_list").set_original("famicom_cass");
 }
 
+void nes_state::famicomo(machine_config &config)
+{
+	famicom(config);
+
+	// basic machine hardware
+	n2a03_device &maincpu(N2A03(config.replace(), m_maincpu, NTSC_APU_CLOCK));
+	maincpu.set_addrmap(AS_PROGRAM, &nes_state::nes_map);
+
+	m_ppu->set_cpu_tag(m_maincpu);
+	m_ppu->int_callback().set_inputline(m_maincpu, INPUT_LINE_NMI);
```

Comment:
Do you actually need these two lines?  Object finders are clever and can deal with `device_replace`,

Reply:
Thanks. Removed. I've seen other things break after a replace, but I guess I was being needlessly defensive.

Reply:
I've definitely seen CPU devices lose their memory map after a replace.  I didn't actually know it was supposed to keep the hookups.

Reply:
The device that is replaced loses everything.  Devices configured to find the replaced device still find it.  In this case, the CPU needs to be reconfigured after being replaced, but the PPU does not need to be reconfigured to find the replacement CPU.